### ISSUE TITLE
`ft-transfer?` allows to steal other people's tokens

### ISSUE THREAD

**Issue by LNow** [blockstack/stacks-blockchain#2775]
**Describe the bug**
`ft-transfer?` allows you to steal other people's tokens if it is not secured by Clarity developer with proper check.

**Steps To Reproduce**
Clone https://github.com/LNow/ft-steal and run tests using `clarinet test` command.
First test is not failing which means I can steal someones tokens if contract developer forgot to secure `ft-transfer` function.

**Expected behavior**
Both tests should fail

**Comment by whoabuddy** [comment]
In finding this, we also discovered this is the one way that `stx-transfer?` and `ft-transfer?` differ, the former throws `(err u4)` in this case.

https://docs.stacks.co/references/language-functions#stx-transfer

```
(err u1) -- sender does not have enough balance to transfer
(err u2) -- sender and recipient are the same principal
(err u3) -- amount to send is non-positive
(err u4) -- the sender principal is not the current tx-sender
```

https://docs.stacks.co/references/language-functions#ft-transfer

```
(err u1) -- sender does not have enough balance to transfer
(err u2) -- sender and recipient are the same principal
(err u3) -- amount to send is non-positive
```

**Comment by MarvinJanssen** [comment]
Did you try this on mocknet/testnet? This would be covered by post conditions. Clarinet should have a post condition mechanism.

**Comment by lgalabru** [comment]
I can see how you got confused, `ft-transfer?` and `stx-transfer?` have similar syntaxes, but are 2 very different concepts.

`stx-transfer?` is a function that allows people transferring STX tokens.
`ft-transfer?` is a **building brick** that developer can use for building contracts, that can act like Token contracts like SIP009/SIP010, or others.

It's the responsibility of the developer to define the notion of ownership, which will differ according the type of contracts you're working on.
For instance, in BNS, you own your name for duration defined by the contract, and when the lease expires, you lose this ownership automatically.

**Comment by lgalabru** [comment]
TLDR: not a bug but a feature, however the documentation should highlight the fact that the design of the ownership (who can transfer, create and burn tokens) is the responsibility of the developer.

**Comment by lgalabru** [comment]
Transferring this issue to the doc repo.

**Comment by LNow** [comment]
I get the concept of having such **building block** and I'm OK with that. Who knows, maybe one day someone will build CBDC on Stacks thanks to this feature :smile: 

Main conclusion is that tokens ownership has to be in https://docs.stacks.co/references/language-functions
Preferably using different examples (full ownership, partial ownership (like BNS), etc.)
